tests/lava/lib/optimization/utils/generators/test_mis.py

```diff
@@ -3,96 +3,67 @@
-                                   [0, 0, 0, 1, 1, 1, 1, 1, 1, 1],
-                                   [1, 0, 0, 1, 1, 0, 1, 1, 1, 1],
-                                   [1, 1, 1, 0, 0, 0, 1, 1, 1, 1],
-                                   [1, 1, 1, 0, 0, 1, 1, 1, 1, 1],
-                                   [1, 1, 0, 0, 1, 0, 1, 1, 1, 1],
-                                   [1, 1, 1, 1, 1, 1, 0, 0, 1, 0],
-                                   [0, 1, 1, 1, 1, 1, 0, 0, 1, 1],
-                                   [1, 1, 1, 1, 1, 1, 1, 1, 0, 1],
-                                   [1, 1, 1, 1, 1, 1, 0, 1, 1, 0]])
-        self.assertTrue((correct_matrix == matrix).all())
+        self.assertEqual(graph.number_of_nodes(), self.num_vertices)
+        self.assertEqual(graph.number_of_edges() * 2, self.num_edges)
 
     def test_get_complement_graph(self):
         """Tests that the complement graph contains the correct number of
         nodes and edges."""
         graph = self.problem.get_complement_graph()
-        number_of_nodes = 10
-        number_of_edges = 8
-        self.assertEqual(graph.number_of_nodes(), number_of_nodes)
-        self.assertEqual(graph.number_of_edges(), number_of_edges)
+        self.assertEqual(graph.number_of_nodes(), self.num_vertices)
+        self.assertEqual(graph.number_of_edges() * 2, self.num_edges)
 
     def test_get_complement_graph_matrix(self):
         """Tests the correct complement graph adjacency matrix is returned."""
         matrix = self.problem.get_complement_graph_matrix()
-        correct_matrix = np.array([[0, 1, 0, 0, 0, 0, 0, 1, 0, 0, ],
-                                   [1, 0, 1, 0, 0, 0, 0, 0, 0, 0, ],
-                                   [0, 1, 0, 0, 0, 1, 0, 0, 0, 0, ],
-                                   [0, 0, 0, 0, 1, 1, 0, 0, 0, 0, ],
-                                   [0, 0, 0, 1, 0, 0, 0, 0, 0, 0, ],
-                                   [0, 0, 1, 1, 0, 0, 0, 0, 0, 0, ],
-                                   [0, 0, 0, 0, 0, 0, 0, 1, 0, 1, ],
-                                   [1, 0, 0, 0, 0, 0, 1, 0, 0, 0, ],
-                                   [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ],
-                                   [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, ]])
-        self.assertTrue((correct_matrix == matrix).all())
+        correct_matrix = np.array(
+            [[0, 0, 0, 1], [0, 0, 1, 1], [0, 1, 0, 0], [1, 1, 0, 0]]
+        )
+        self.assertTrue((matrix == correct_matrix).all())
 
     def test_get_as_qubo(self):
         """Tests the conversion to QUBO returns the correct cost matrix."""
-        w_diag = 1.0
-        w_off = 4.0
+        w_diag = 1
+        w_off = 4
         qubo = self.problem.get_as_qubo(w_diag, w_off)
-        self.assertTrue(np.all(qubo.q[np.diag_indices(10)] == -w_diag))
+        self.assertTrue(np.all(qubo.q[np.diag_indices(4)] == -w_diag))
         self.assertTrue(np.all(qubo.q[qubo.q > 0] == w_off / 2))

```

Comment:
Could you add an assert here to ensure that the QUBO matrix is symmetric? Thanks